### DISCUSSION TITLE
How can I freeze first row?

### DISCUSSION THREAD

**Discussion by phaibin** [exceljs/exceljs#2514] [Q&A]

**Comment by guyonroche** [comment]
Like this:
worksheet.views = [
    {state: 'frozen', ySplit: 1}
];

**Comment by zjleon** [comment]
I tried the newest code, and found the 'views' is moved to options, but the document is not updated accordingly, so anyone use the newest version should try below code:
```
workbook.addWorksheet(sheetName, {
views: [
{state: 'frozen', ySplit: 1}
]
});
```

**Comment by marko033** [comment]
What if i want freeze not first but only second row.
When i use ySplit: 2 will be frozen first and second row but i want only second not first.
Is this possible?

**Comment by nakshay** [comment]
> What if i want freeze not first but only second row.
> When i use ySplit: 2 will be frozen first and second row but i want only second not first.
> Is this possible?

I have same question, how to freeze selected cells

**Comment by johnGeorge4** [comment]
> > What if i want freeze not first but only second row.
> >When i use ySplit: 2 will be frozen first and second row but i want only second not first.
> >Is this possible?

> I have same question, how to freeze selected cells

Has anyone been able to find a solution? I'm trying this as well, and even when setting up a template that I am filling out through my code I can't find a solution or viable work-around.

**Comment by nakshay** [comment]
> > > What if i want freeze not first but only second row.
> > > When i use ySplit: 2 will be frozen first and second row but i want only second not first.
> > > Is this possible?
> 
> > I have same question, how to freeze selected cells
> 
> Has anyone been able to find a solution? I'm trying this as well, and even when setting up a template that I am filling out through my code I can't find a solution or viable work-around.

didn't find anything useful yet

**Comment by gergoh** [comment]
Any update on freezing other rows than the first? I've been searching for a while now.

**Comment by johnGeorge4** [comment]
@gergoh I never found a solution. Luckily for me, I was able to convince my team that we needed to find another design to go with.

**Comment by gergoh** [comment]
@johnGeorge4 Seems like, I need to do the same thing. ExcelJS seems more and more limited as I work with it. It's really unfortunate.

**Comment by johnGeorge4** [comment]
@gergoh honestly from the research I did while working on this at the time, it's more of a limitation of Excel itself. If I recall correctly, Excel is capable of freezing the first row (or multiple rows, given that there are no unfrozen rows between the first row and the end of the frozen rows) or freezing columns (though I did not look much into this part of the feature since I had no use for it). If there is ever an update to Excel to allow for more dynamic freezing of rows, then I wouldn't be surprised to see an update to ExcelJS to account for it. It seems right now the biggest roadblock is the default software that the package is using, if that makes sense.

**Comment by anaspsi** [comment]
I still did not receive an answer
